feat(client): Add navigation to credit dashboard

bruceharris opened the pull request:
- currently only supported in web browser, not in OpenFin or other platforms
- `DropdownMenu.tsx` is a generalized implementation of `CurrencyOptions.tsx`, which as an aside, is
  - tightly coupled to the currency use case
  - [unused at present, as far as I can tell](https://github.com/AdaptiveConsulting/ReactiveTraderCloud/blob/master/src/client/src/App/LiveRates/MainHeader/CurrencyOptions.tsx#L13)
- This PR currently targets the `upgrade-openfin` branch, but #2062 should be merged first, and once merged, we should change the base of this PR to `master`
- set `VITE_IS_CREDIT_ENABLED=yes` in `.env.local` to enable the feature

![nav-browser](https://user-images.githubusercontent.com/1081979/169888731-7d6db201-9d8d-4f13-9fff-6a1dc0725e2c.gif)

taylor-adaptive commented:
The work around routes and feature toggling all looks fine.
The `CurrencyOptions` is used - It's hidden but shown at smaller screen sizes/mobile.
I'd suggest keeping the `DropdownMenu` component you've created and refactor `CurrencyOptions` to use it so we're not duplicating it. 
After that, this is good to go.

bruceharris commented:
Thanks @taylor-adaptive - I removed `CurrencyOptions` in favor of `DropdownMenu`, and also fixed a bug where the state of the currency selection state was previously not synchronized between the 2 currency selection components.

## Before:
![currency options bug](https://user-images.githubusercontent.com/1081979/170561868-87ffe0e1-4a37-40bf-b463-bdaee4f628be.gif)

## After fix:
![currency options bug fixed](https://user-images.githubusercontent.com/1081979/170561859-d4ba6c11-4e24-494f-a8f8-d78ba4f3dd84.gif)

taylor-adaptive commented on src/client/src/theme/ThemeContext.tsx:
```diff
@@ -73,7 +73,7 @@ export const ThemeProvider: React.FC<Props> = ({
 }
 
-export const useTheme = () => {
+export const useThemeName = () => {
```
What was your reasoning behind this name change? The hook returns more than just the theme name.

bruceharris replied:
Noted in commit message of - https://github.com/AdaptiveConsulting/ReactiveTraderCloud/pull/2063/commits/9baeda8c1ff384456174c5cbfbee23275a159e01 - to avoid confusion with
https://styled-components.com/docs/api#usetheme which returns a theme object and is used elsewhere in the app, e.g. https://github.com/AdaptiveConsulting/ReactiveTraderCloud/blob/master/src/client/src/OpenFin/Window/LayoutLock.tsx#L3

Open to suggestions for a better name, or we can just drop this commit if the original is preferred.

taylor-adaptive replied:
It doesn't bother me too much, I think we probably have other instances of similar names with dependencies.
If you feel strongly about it then maybe something like `useApplicationTheme` or `useReactiveTraderTheme`, or drop the change.

taylor-adaptive commented:
> Thanks @taylor-adaptive - I removed `CurrencyOptions` in favor of `DropdownMenu`, and also fixed a bug where the state of the currency selection state was previously not synchronized between the 2 currency selection components.
> 
> ## Before:
> ![currency options bug](https://user-images.githubusercontent.com/1081979/170561868-87ffe0e1-4a37-40bf-b463-bdaee4f628be.gif) [ ](https://user-images.githubusercontent.com/1081979/170561868-87ffe0e1-4a37-40bf-b463-bdaee4f628be.gif) ![currency options bug](https://user-images.githubusercontent.com/1081979/170561868-87ffe0e1-4a37-40bf-b463-bdaee4f628be.gif) [ ](https://user-images.githubusercontent.com/1081979/170561868-87ffe0e1-4a37-40bf-b463-bdaee4f628be.gif)
> 
> ## After fix:
> ![currency options bug fixed](https://user-images.githubusercontent.com/1081979/170561859-d4ba6c11-4e24-494f-a8f8-d78ba4f3dd84.gif) [ ](https://user-images.githubusercontent.com/1081979/170561859-d4ba6c11-4e24-494f-a8f8-d78ba4f3dd84.gif) ![currency options bug fixed](https://user-images.githubusercontent.com/1081979/170561859-d4ba6c11-4e24-494f-a8f8-d78ba4f3dd84.gif) [ ](https://user-images.githubusercontent.com/1081979/170561859-d4ba6c11-4e24-494f-a8f8-d78ba4f3dd84.gif)

Looks good!

bruceharris commented:
@TobyLi96  @taylor-adaptive removed the `useTheme` refactor commit and retargeted the PR to master
